### ISSUE TITLE
Batched Transaction Ignores Recipients

### ISSUE THREAD

**Issue by PeteClubSeven** [nunchuk-io/libnunchuk#27]
Hi, I'm raising this bug in this repository because I can't see an iOS repository. If you'd prefer it in another I'm happy to move it.

I found a really bad bug with the batched transaction feature on iOS (it does not effect macOS). If you try to customize the transaction before creating it then only the first recipient will be used in the final transaction and the total amount of sats you were trying to send will go to the first recipient. I had this happen before and thought I had made a mistake but it's actually a bug with the app. Here's a real world example of it happening:

1. Generate two receive addresses in Nunchuk
2. Open a wallet in Nunchuk on iOS
3. Press "Send"
4. Press the batch icon in the top-right corner of the app
5. Enter/scan the first recipient address
6. Enter "123" sats for the first recipient
7. Enter/scan the second recipient address
8. Enter "456" sats for the second recipient address
9. Press "Customize transaction" at the bottom of the screen
10. Press "Continue" at the bottom of the screen
11. Note how there are two addresses shown in the "Send to address" section with 123 and 456 sats respectively, everything is normal
12. Press "Confirm and create transaction"
13. Press "More details"
14. Notice how there is only one address shown in the "Send to" section
15. Notice how 579 sats are being sent to the address

If you sign and broadcast this transaction it will be constructed in the same manner shown with only one address receiving all of the satoshi's you previously specified. It's also worth noting that if you DO NOT choose to customize the transaction then it will be formed exactly as the user expects it to be. This is a really bad bug which has caused me issues before.

Batching Screen:
![Screenshot 2023-11-01 at 16 10 26](https://github.com/nunchuk-io/libnunchuk/assets/130258664/9e9c0dd5-726f-4bc4-8f3d-48b0c9dfd02c)

Confirmation Screen:
![Screenshot 2023-11-01 at 16 13 10](https://github.com/nunchuk-io/libnunchuk/assets/130258664/210eec5d-ca30-41cc-a28c-87b444d5778f)

Final Transaction Screen:
![Screenshot 2023-11-01 at 16 16 42](https://github.com/nunchuk-io/libnunchuk/assets/130258664/040fb3ff-ae6e-4239-bc10-856504635d01)

**Comment by hugohn** [comment]
Looks like an iOS bug. Thanks @PeteClubSeven .

**Comment by PeteClubSeven** [comment]
> Looks like an iOS bug. Thanks @PeteClubSeven .

Yeah it doesn't happen on the macOS desktop version but I didn't test the Android version. Is there a separate iOS repository? I didn't see one on the organization page.
